lib/dpp/isolation/IsolatedDataProviderWrapper.js

```diff
@@ -0,0 +1,130 @@
+// This class wraps external reference to data provider for the isolate
+const DashPlatformProtocol = require('@dashevo/dpp');
+const Document = require('@dashevo/dpp/lib/document/Document');
+const { Transaction } = require('@dashevo/dashcore-lib');
+
+const DATA_PROVIDER_METHODS = {
+  FETCH_DATA_CONTRACT: 'fetchDataContract',
+  FETCH_IDENTITY: 'fetchIdentity',
+  FETCH_DOCUMENTS: 'fetchDocuments',
+  FETCH_TRANSACTION: 'fetchTransaction',
+};
+
+class IsolatedDataProviderWrapper {
+  /**
+   * @param {DataProvider} dataProviderExternalReference
+   */
+  constructor(dataProviderExternalReference) {
+    this.internalDpp = new DashPlatformProtocol();
+    this.dataProviderExternalReference = dataProviderExternalReference;
+  }
+
+  async callMethodFromReference(methodName, ...args) {
+    const methodReference = await this.dataProviderExternalReference.get(methodName);
+
+    return methodReference.apply(
+      this.dataProviderExternalReference.derefInto(),
+      args,
+      { arguments: { copy: true }, result: { promise: true } },
+    );
+  }
+
+  /**
+   * Fetch Contract by ID
+   *
+   * @param {string} id
+   * @return {Promise<DataContract|null>}
+   */
+  async fetchDataContract(id) {
+    const dataContractReference = await this.callMethodFromReference(
+      DATA_PROVIDER_METHODS.FETCH_DATA_CONTRACT, id,
+    );
+    if (!dataContractReference) {
+      return null;
+    }
+
+    const toJSONReference = await dataContractReference.get('toJSON');
+
+    const dataContractJSON = await toJSONReference.apply(
+      dataContractReference.derefInto(),
+      [],
+      { result: { copy: true } },
+    );
+
+    return this.internalDpp.dataContract.createFromObject(
```

Comment:
In the next major DPP version, we gonna change `DataContract` constructor in order to accept `rawDataContract` so we will be able to remove internal DPP.